tests/functional/conftest.py

```diff
@@ -191,6 +191,33 @@ def _wait(timeout=30, step=0.5):
     return _wait
 
 
+def wait_for_deleted(
```

Comment:
In general, this now duplicates the deletion logic in all our fixtures where we do delete and then call this. So I would remove those deletes and add this as the only finalizer and maybe call it e.g. `safe_delete` or so?